rllib/env/multi_agent_episode.py

```diff
@@ -796,10 +803,10 @@ def cut(self, len_lookback_buffer: int = 0) -> "MultiAgentEpisode":
                 "Can't call `MultiAgentEpisode.cut()` when the episode is already done!"
             )
 
-        # If there is data (e.g. actions) in the agents' buffers, we might have to
-        # re-adjust the lookback len further into the past to make sure that these
+        # If there is hanging data (e.g. actions) in the agents' caches, we might have
+        # to re-adjust the lookback len further into the past to make sure that these
         # agents have at least one observation to look back to.
-        for agent_id, agent_actions in self._agent_buffered_actions.items():
+        for agent_id, agent_actions in self._hanging_actions_end.items():
             assert self.env_t_to_agent_t[agent_id].get(-1) == self.SKIP_ENV_TS_TAG
```

Comment:
Is this always true? What if an agent stepped in the last timestep but did not receive an observation? Do we then denote this as no-step (`SKIP_ENV_TS_TAG`) and count the step at the timestep the missing observation comes in?

Reply:
correct.
As long as the agent does NOT receive a new obs, all steps get marked with "SKIP_ENV_TS_TAG".